### PR TITLE
Remove non inclusive text

### PR DESCRIPTION
**Pull request by fossedihelm**
**What this PR does / why we need it**:
Remove some non-inclusive terms from the codebase and fix some broken links in documentation.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
This PR contains changes already reviewed in the PR [#6050](https://github.com/kubevirt/kubevirt/pull/6050). I cherry-picked that commits, applied the requested changes and added others.

**Release note**:

```release-note
Changed some terminology to be more inclusive
```

### CODE REVIEW

**Comment by fossedihelm** [comment]
/cc @rmohr @jean-edouard 
I add to you, because you made the review to the previous linked PR. Thanks

**Comment by rmohr** [comment]
/test all

@fossedihelm looks great. Let's check that the code changes are sane.

File: docs/devel/storage.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ After a few moments the hostpath-provisioner pod should be running, we are now r
 The Containerized Data Importer controller is a controller that watches for PVCs with a specific annotation. If that annotation is detected, it will use the 'storage.image.endpoint' URL to download the image, convert it if needed and write it to the requested PVC. One can then use the PVC to start a VM using the image in it. To deploy the CDI controller run this:
 
 ```bash
-cluster-up/kubectl.sh apply -f https://raw.githubusercontent.com/kubevirt/containerized-data-importer/master/manifests/controller/cdi-controller-deployment.yaml
+cluster-up/kubectl.sh apply -f https://raw.githubusercontent.com/kubevirt/containerized-data-importer/main/manifests/controller/cdi-controller-deployment.yaml
```
Comment (jean-edouard) [inline docs/devel/storage.md:39]: 404

File: docs/reviewer-guide.md [hunk 2]
```diff
@@ -25,7 +25,7 @@ Make a few passes over the code you want to review.
 
 ## Pull Request structure
 
-* It's preferred that authors rebase on master instead of merging master into their PRs.
+* It's preferred that authors rebase on main instead of merging main into their PRs.
```
Comment (jean-edouard) [inline docs/reviewer-guide.md:28]:
```suggestion
* It's preferred that authors rebase on main instead of merging the main branch into their PRs.
```

**Review by jean-edouard — COMMENTED** [review]
/ok-to-test

File: docs/devel/storage.md (hunk 1 above)
Comment (fossedihelm) [reply docs/devel/storage.md:39]: Thanks @jean-edouard. I'm currently working on how to adjust the documentation. The file has been removed in the PR [#398](https://github.com/kubevirt/containerized-data-importer/pull/398).

File: docs/reviewer-guide.md (hunk 2 above)
Comment (fossedihelm) [reply docs/reviewer-guide.md:28]: Thanks, sounds better.

**Comment by jean-edouard** [comment]
@fossedihelm did you remove `docs/devel/storage.md` on purpose?

**Comment by fossedihelm** [comment]
> @fossedihelm did you remove `docs/devel/storage.md` on purpose?
Yes I was replying in the previous comment. As suggested by @awels the file has been deleted as none of it is relevant anymore.

**Comment by awels** [comment]
yes docs/devel/storage.md is ancient and no longer relevant in any capacity. This was relevant years ago, but not now.

**Pull request merged**